src/main/java/entities/post.java

```diff
@@ -0,0 +1,130 @@
+     * @param replies the comments that are added to this post. This is an attribute of the superclass Postable.
+     * @param numPostsCreated1 the number of posts created so far.
+     */
+    public Post(User poster, String title, String mainDesc, List<String> tags, String collaborators,
+                Date deadline, List<Comment> replies, int numPostsCreated1){
+        super.user = poster;
+        super.body = mainDesc;
+        super.replies = replies;
+        numPostsCreated = numPostsCreated1;
+        numPostsCreated++;
+        this.id = numPostsCreated;
+        this.title = title;
+        this.tags = tags;
+        this.collaborators = collaborators;
+        this.deadline = deadline;
+    }
+
+    /**
+     * @return the Post's ID.
+     */
+    public int getID(){
+        return this.id;
+    }
+
+    /**
+     * @return a list of the Users who favourited the Post.
+     */
+    public List<User> getFavouritedUsers(){
+        return this.favouritedUsers;
+    }
+
+    /**
+     * @return a list of tags specified by the User who made the Post.
+     */
+    public List<String> getTags(){
+        return this.tags;
+    }
+
+    /**
+     *
+     * @return the number of Users who favourited the Post.
+     */
+    public int getNumFavouritedUsers(){
+        return this.favouritedUsers.size();
+    }
+
+    /**
+     * adds a User to the list of Users who favourited the Post if it is not in favouritedUsers yet.
+     * @param favouritedUser the User who favourited the Post.
+     * @return true if user has been added; false otherwise.
+     */
+    public boolean addFavouritedUser(User favouritedUser){
+        if(!(favouritedUsers.contains(favouritedUser))){
+            this.favouritedUsers.add(favouritedUser);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Searches for userToRemove in the list of Users who favourited the Post, removes userToRemove if
+     * that user is there.
+     * @param userToRemove the User to be removed.
+     * @return true if the user has been removed; false otherwise.
+     */
+    public boolean removeFavouritedUser(User userToRemove){
+        for(int i = 0; i < favouritedUsers.size(); i++){
+            if(userToRemove.equals(favouritedUsers.get(i))){
+                favouritedUsers.remove(i);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * set the tags that this Post is categorised as by the Poster.
+     * @param tags the tags this Post is categorised as by the Poster.
+     */
+    public void setTags(List<String> tags){
+        this.tags = tags;
+    }
+
+    /**
+     * Checks whether two posts are equal based on whether their Post id's are equal.
+     * @param o the Post to be compared to.
+     * @return a boolean that reflects whether the two Posts' id's are equal.
+     */
+    @Override
+    public boolean equals(Object o){
+        if(this == o){
```

Comment:
Correct me if I'm wrong but isn't this a redundant check? I think this is covered by the check at the end of the method.

Reply:
I think you're right! It's just that they implemented it this way in the class notes so I copied it; I am not exactly sure about the reason though.